src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -198,6 +198,7 @@ public boolean topicExists(final String topic) throws TopicRepositoryException {
             item.updateStatusAndDetail(EventPublishingStatus.FAILED, "internal error");
             throw new EventPublishingException("Error publishing message to kafka", e);
         } catch (final RuntimeException e) {
+            kafkaFactory.terminateProducer(producer);
```

Comment:
I thought that we should do that when kafka reports about exception, not (which it detected by listener in producer.send())

Reply:
Sorry, ignore this comment, I was wrong

Reply:
@antban can you provide details regarding this? Because it was also not clear for me why we terminate producer by getting any RuntimeException.